#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *
#from local_planner_limits import add_generic_localplanner_params

gen = ParameterGenerator()

# This unusual line allows to reuse existing parameter definitions
# that concern all localplanners
#add_generic_localplanner_params(gen)

# For integers and doubles:
#       Name                    Type      Reconfiguration level
#       Description
#       Default  Min  Max

gen.add("enable_config", bool_t, 0, "Enable to update parameter",  True)


gen.add("inflation_radius", double_t, 0, 
                            "inflation radius for obstacles", 
                            0.3, 0.01, 1.0)

gen.add("min_distance_between_points", double_t, 0, 
                            "minimum distance between two vertices in different polygon obstacles", 
                            0.1, 0.01, 1.0)
exit(gen.generate("obstacle_detection", "obstacle_detection", "ObstacleDetectionReconfigure"))
